rosetta-lang/src/main/java/com/regnosys/rosetta/types/ExpectedTypeProvider.java

```diff
@@ -84,11 +84,15 @@
 public interface ExpectedTypeProvider {	
 	static final int INSIGNIFICANT_INDEX = -1;
 	
-	RType getExpectedTypeFromContainer(EObject owner);
-	default RType getExpectedType(EObject owner, EReference reference) {
+	RMetaAnnotatedType getExpectedTypeFromContainer(EObject owner);
+	default RMetaAnnotatedType getExpectedType(EObject owner, EReference reference) {
 		return getExpectedType(owner, reference, INSIGNIFICANT_INDEX);
 	}
-	RType getExpectedType(EObject owner, EReference reference, int index);
+	RMetaAnnotatedType getExpectedType(EObject owner, EReference reference, int index);
+	
+	private static RMetaAnnotatedType withEmptyMeta(RType rType) {
+		return new RMetaAnnotatedType(rType, List.of());
+	}
```

Comment:
Would be better to have as a public static factory method on `RMetaAnnotatedType`

Reply:
done